packages/plugin-ext/src/main/node/plugin-service.ts

```diff
@@ -37,20 +50,32 @@ export class PluginApiContribution implements BackendApplicationContribution {
 
         const webviewApp = connect();
         webviewApp.use('/webview', serveStatic(path.join(__dirname, '../../../src/main/browser/webview/pre')));
-        const webviewExternalEndpoint = this.webviewExternalEndpoint();
-        console.log(`Configuring to accept webviews on '${webviewExternalEndpoint}' hostname.`);
-        app.use(vhost(new RegExp(webviewExternalEndpoint, 'i'), webviewApp));
+        app.use(vhost(this._webviewExternalEndpointRegExp, webviewApp));
+    }
+
+    allowWsUpgrade(request: http.IncomingMessage): MaybePromise<boolean> {
```

Comment:
what for? I don't think we allow such requests in service worker already

Reply:
See https://bugs.eclipse.org/bugs/show_bug.cgi?id=568018#c3

Reply:
And https://stackoverflow.com/a/39602235/7983255

Reply:
ok, I still not sure what additional harm can VS Code extension do here. It already has full access on the backend.

Reply:
It is not to prevent malicious extensions, but to guard against known vectors that could originate from other origins.

See the mini-browser example from the bugzilla: Just changing its origin is not enough to prevent it, we need to also check.